src/routes/NewBoard/NewBoard.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { useFirestore } from 'react-redux-firebase';
+import Input from '@material-ui/core/Input';
+import Button from '@material-ui/core/Button';
+import {AuthenticationManager} from 'utils/authentication/AuthenticationManager';
+import BoardData from "types/BoardData";
+import firebase from 'firebaseSetup';
+
+import getRandomName from 'constants/Name';
+import {RouteComponentProps} from "react-router";
+
+interface NewBoardProps extends RouteComponentProps {}
+
+function NewBoard(props: NewBoardProps) {
+
+    const firestore = useFirestore();
+
+    let auth = firebase.auth().currentUser;
+
+    const [name, setName] = React.useState(getRandomName());
+    function handleChangeName(e: any) {
+        const name = (e.target as HTMLInputElement).value;
+        setName(name);
+    }
+
+    function handleLogin() {
+        AuthenticationManager.signInAnonymously(name).then(_ => {
+            auth = firebase.auth().currentUser;
+            createBoard();
+        });
+    }
+
+    function createBoard() {
+        const newBoard: BoardData = {
+            owner: auth!.uid,
+            topic: "Test",
+            date: (firestore.Timestamp as any).fromDate(new Date()),
+            config: {
+                private: true,
+                encryption: false,
+                maxVotes: 5,
+                multiVote: true,
+                hideVotes: false
+            }
+        };
+
+        firestore.collection('boards').add(newBoard).then(success => {
+            firestore.collection('boards').doc(success.id).collection('participants').doc(auth!.uid).set({
+                name: auth!.displayName,
+                admin: true
+            });
+            props.history.push('/board/' + success.id);
```

Comment:
I would prefer template literals and write `/board/${success.id}` here. In my opinion it is much easier to read (... with syntax highlighting in any IDE)